pkg/shares/counter_test.go

```diff
@@ -0,0 +1,72 @@
+package shares_test
+
+import (
+	"bytes"
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	coretypes "github.com/tendermint/tendermint/types"
+
+	"github.com/celestiaorg/celestia-app/pkg/appconsts"
+	"github.com/celestiaorg/celestia-app/pkg/namespace"
+	"github.com/celestiaorg/celestia-app/pkg/shares"
+)
+
+func TestCounterMatchesCompactShareSplitter(t *testing.T) {
+	testCases := []struct {
+		txs []coretypes.Tx
+	}{
+		{txs: []coretypes.Tx{newTx(120)}},
+		{txs: []coretypes.Tx{newTx(appconsts.FirstCompactShareContentSize - 2)}},
+		{txs: []coretypes.Tx{newTx(appconsts.FirstCompactShareContentSize - 1)}},
+		{txs: []coretypes.Tx{newTx(appconsts.FirstCompactShareContentSize), newTx(appconsts.ContinuationCompactShareContentSize - 4)}},
```

Comment:
[optional] more test cases that pass

```suggestion
		{txs: []coretypes.Tx{newTx(appconsts.FirstCompactShareContentSize)}},
		{txs: []coretypes.Tx{newTx(appconsts.FirstCompactShareContentSize + 1)}},
		{txs: []coretypes.Tx{newTx(appconsts.FirstCompactShareContentSize), newTx(appconsts.ContinuationCompactShareContentSize - 4)}},
```